airflow_tecton/operators/job_utils.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 Tecton, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import pprint
+import time
+
+from airflow_tecton.hooks.tecton_hook import TectonHook
+
+
+def wait_until_completion(hook: TectonHook,
+                          workspace: str,
+                          feature_view: str,
+                          job_id: str):
+    """
+    Wait materialization job to finish successfully.
+    If the job fails, throw exception.
+    """
+    job_result = hook.get_materialization_job(
+        workspace, feature_view, job_id
+    )["job"]
+    while job_result["state"].upper().endswith("RUNNING"):
+        if "attempts" in job_result:
```

Comment:
i know this is copy paste, but on second though, should probably add the condition `add job_result['attempts']` in case the logic somehow changes and we return empty lists in some cases